packages/lib-classifier/src/store/Subject.spec.js

```diff
@@ -57,6 +60,66 @@ describe('Model > Subject', function () {
+      const multiFrameSubject = SubjectFactory.build({ locations: [
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' }
+      ]})
+      const subjectStore = Subject.create(multiFrameSubject)
+      subjectStore.workflows = WorkflowStore.create({})
+      subjectStore.workflows.setResource(workflow)
+      subjectStore.workflows.setActive(workflow.id)
+      expect(subjectStore.viewer).to.equal(subjectViewers.multiFrame)
+    })
+
+    it('should return a null viewer for subjects with more than ten location', function () {
+      const multiFrameSubject = SubjectFactory.build({ locations: [
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+        { 'image/png': 'https://foo.bar/example.png' },
+      ]})
+      const subjectStore = Subject.create(multiFrameSubject)
+      subjectStore.workflows = WorkflowStore.create({})
+      subjectStore.workflows.setResource(workflow)
+      subjectStore.workflows.setActive(workflow.id)
+      expect(subjectStore.viewer).to.be.null()
+    })
+
+    it('should return a null viewer when workflow.configuration["multi_image_mode"] === "separate"', function () {
+      const multiFrameSubject = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }, { 'image/png': 'https://foo.bar/example.png' }]})
+      const subjectStore = Subject.create(multiFrameSubject)
+      subjectStore.workflows = WorkflowStore.create({})
+      subjectStore.workflows.setResource(workflowWithConfigSeparateMultiImage)
+      subjectStore.workflows.setActive(workflowWithConfigSeparateMultiImage.id)
+      expect(subjectStore.viewer).to.be.null()
+    })
+
+    it('should return a null viewer when workflow.configuration["enable_switching_flipbook_and_separate"] === "true"', function () {
+      const multiFrameSubject = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }, { 'image/png': 'https://foo.bar/example.png' }]})
+      const subjectStore = Subject.create(multiFrameSubject)
+      subjectStore.workflows = WorkflowStore.create({})
+      subjectStore.workflows.setResource(workflowWithConfigEnableSwitching)
+      subjectStore.workflows.setActive(workflowWithConfigEnableSwitching.id)
+      expect(subjectStore.viewer).to.be.null()
+    })
+
+    it('should return the multiFrame viewer if the workflow configuration for subject_viewer is defined as multiFrame', function () {
```

Comment:
This is to test that regardless of the subject, including type or number of locations, if the workflow config is set to use multi frame viewer then the subject uses the multi frame viewer? Nice!

Reply:
It is, however, I am realizing there is an assumption that `{ subject_viewer: 'multiFrame' }` can only be enabled for subjects with between 2 and 10 frames.

Reply:
I will need to revisit this logic when creating a workflow config option in the lab editor. Thank you for the question!

Reply:
@beckyrother is working on some preliminary ideas for the workflow lab editor page with these new configurations.